cfme/containers/image.py

```diff
@@ -1,64 +1,64 @@
 # -*- coding: utf-8 -*-
 # added new list_tbl definition
+from navmazing import NavigateToSibling, NavigateToAttribute
+
 from cfme.common import SummaryMixin, Taggable
 from cfme.fixtures import pytest_selenium as sel
-from cfme.web_ui import toolbar as tb, CheckboxTable
-from cfme.web_ui.menu import nav
-from . import details_page
+from cfme.web_ui import toolbar as tb, CheckboxTable, paginator, summary_title, InfoBlock
+from utils.appliance.endpoints.ui import CFMENavigateStep, navigator, navigate_to
+from utils.appliance import Navigatable
 
 list_tbl = CheckboxTable(table_locator="//div[@id='list_grid']//table")
 
-nav.add_branch(
-    'containers_images',
-    {
-        'containers_image':
-        lambda ctx: list_tbl.select_row_by_cells(
-            {'Name': ctx['image'].name, 'Provider': ctx['image'].provider.name}),
 
-        'containers_image_detail':
-        lambda ctx: list_tbl.click_row_by_cells(
-            {'Name': ctx['image'].name, 'Provider': ctx['image'].provider.name}),
-    }
-)
+def match_title_and_controller():
+    title_match = sel.execute_script('return document.title;') == 'CFME: Images'
+    controller_match = sel.execute_script('return ManageIQ.controller;') == 'container_image'
+    return title_match and controller_match
 
 
-class Image(Taggable, SummaryMixin):
+class Image(Taggable, SummaryMixin, Navigatable):
 
-    def __init__(self, name, provider):
+    def __init__(self, name, provider, appliance=None):
         self.name = name
         self.provider = provider
-
-    def _on_detail_page(self):
-        return sel.is_displayed(
-            '//div//h1[contains(., "{} (Summary)")]'.format(self.name))
+        Navigatable.__init__(self, appliance=appliance)
 
     def load_details(self, refresh=False):
-        if not self._on_detail_page():
-            self.navigate(detail=True)
-        elif refresh:
-            tb.refresh()
-
-    def click_element(self, *ident):
-        self.load_details(refresh=True)
-        return sel.click(details_page.infoblock.element(*ident))
+        navigate_to(self, 'Details')
```

Comment:
There is a dynamic usage (cfme.common.Summary) that I do not believe it is worth working around at the moment. When navmazing is more complete, it will be simple to remove.

Added a TODO and replaced refresh.